### PR TITLE
STYLE: Resolved Inconsistent namespace

### LINKED ISSUE #39992: STYLE Increase coverage of inconsistent-namespace-usage
If you already have experience contributing, then please leave this issue to newcomers - thanks :+1: 

---

In `.pre-commit-config.yaml`, the check `inconsistent-namespace-usage` is limited to `pandas/tests/frame/`. Let's expand it to the rest of `pandas/tests/`.

The task here is:

0. make sure you're familiar with the [contributing guide](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html)
1. choose a folder from the list below (e.g. `series`) and add it to the check in `.pre-commit-config.yaml`. E.g., if you chose `series`, that would be:

  ```yaml
      -   id: inconsistent-namespace-usage
          name: 'Check for inconsistent use of pandas namespace in tests'
          entry: python scripts/check_for_inconsistent_pandas_namespace.py
          language: python
          types: [python]
          files: ^pandas/tests/(frame|series)/
  ```

2. temporarily add `tokenize-rt` as an additional dependency and `--replace` as an arg (these will be removed later):

  ```yaml
      -   id: inconsistent-namespace-usage
          name: 'Check for inconsistent use of pandas namespace in tests'
          entry: python scripts/check_for_inconsistent_pandas_namespace.py
          language: python
          types: [python]
          files: ^pandas/tests/(frame|series)/
          additional_dependencies: [tokenize-rt]
          args: [--replace]
  ```

3. run this check on all files with pre-commit, i.e.:

```console
  $ pre-commit run inconsistent-namespace-usage --all-files
  [INFO] Initializing environment for local:tokenize-rt.
  [INFO] Installing environment for local.
  [INFO] Once installed this environment will be reused.
  [INFO] This may take a few minutes...
  Check for inconsistent use of pandas namespace in tests.......................Failed
  - hook id: inconsistent-namespace-usage
  - files were modified by this hook
  ```

4. run it again - this time, it should pass

  ```console
  $ pre-commit run inconsistent-namespace-usage --all-files
  Check for inconsistent use of pandas namespace in tests.......................Passed
  ```
  
  Check that the other hooks still pass - if you enable pre-commit (`pre-commit install`), then this will happen automatically when you try to commit your changes (see step 6)

5. revert the changes to `.pre-commit-config.yaml` (we'll update it once these changes are all so as to avoid a barrage of merge conflicts)
6. commit your changes, and open a pull request. Check over your changes, make sure they look sensible, let me know if not

---

Here are the folders in `pandas/tests` which this check needs expanding to:

- [x] series
- [x] scalar
- [x] resample
- [x] util
- [x] tseries
- [x] dtypes
- [x] arrays
- [x] plotting
- [x] tools
- [x] indexing
- [x] strings
- [x] reductions
- [x] frame
- [x] arithmetic
- [x] config
- [x] api
- [x] generic
- [x] io
- [x] base
- [x] window
- [x] internals
- [x] reshape
- [x] computation
- [x] apply
- [x] groupby
- [x] tslibs
- [x] libs
- [x] indexes
- [x] extension

No need to ask for permission to work on this, just leave a comment letting people know which folder(s) from the list above you're working on

### PR DESCRIPTION
**Pull request by deepang17**
- [x] closes #39992 
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
Hi @deepang17 - instead of excluding a file, can we change `test_unique_tuples` so the last argument of the function is `uniques` instead of `unique`?

**Comment by deepang17** [comment]
Sure, will do that.

**Comment by MarcoGorelli** [comment]
Thanks, looks good - I'll just mark this as draft pending on https://github.com/pandas-dev/pandas/issues/39992#issuecomment-792294708 , will then reopen and rebase (depending on what we hear back)

**Comment by MarcoGorelli** [comment]
Hi @deepang17 - #40310 is in, do you want to rebase and fix up any remaining errors so we can get this in?

**Comment by deepang17** [comment]
Hello, @MarcoGorelli I have completed work from my side. If you think that changing from `unique` to `uniques` is not a breaking change then this works fine.

**Comment by MarcoGorelli** [comment]
You'll also need to change it to `uniques` in
```
    @pytest.mark.parametrize(
        "arr, unique",
```

It's just a variable in a test, so it's not a breaking change

If you could fetch and merge upstream/master and fixup "unique" in `pytest.mark.parametrize` then this should be good to go

**Comment by deepang17** [comment]
Ok, will do that.

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks for updating - there's still some files to be fixed up though, which are failing the pre-commit check

**Comment by deepang17** [comment]
image

@MarcoGorelli these new files are producing the error.

image

**Comment by deepang17** [comment]
Shall I convert them to arrays?

**Comment by MarcoGorelli** [comment]
I'd suggest renaming to `arr` rather than `arrays`, as here there's only one of them (`uniques` in `test_algos.py` is fine as there's many objects there anyway many objects and arguably might have been better named `uniques` from the start)

**Comment by deepang17** [comment]
`arr` is used at quite a few places, so it will again create conflicts. Please suggest something else. Thank you.

File: pandas/tests/arithmetic/test_interval.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def left_right_dtypes(request):
 
 
 @pytest.fixture
-def array(left_right_dtypes):
+def arrays(left_right_dtypes):
```
Comment (MarcoGorelli) [inline pandas/tests/arithmetic/test_interval.py:53]: can we rename to `interval_array`?

File: pandas/tests/arrays/sparse/test_array.py [hunk 2]
```diff
@@ -522,7 +522,7 @@ def test_astype_all(self, any_real_dtype):
         tm.assert_numpy_array_equal(np.asarray(res.to_dense()), vals.astype(typ))
 
     @pytest.mark.parametrize(
-        "array, dtype, expected",
+        "arrays, dtype, expected",
```
Comment (MarcoGorelli) [inline pandas/tests/arrays/sparse/test_array.py:525]: how about `sparse_array`?

File: pandas/tests/arrays/string_/test_string.py [hunk 3]
```diff
@@ -223,31 +223,31 @@ def test_mul(dtype, request):
 
 @pytest.mark.xfail(reason="GH-28527")
 def test_add_strings(dtype):
-    array = pd.array(["a", "b", "c", "d"], dtype=dtype)
+    arrs = pd.array(["a", "b", "c", "d"], dtype=dtype)
```
Comment (MarcoGorelli) [inline pandas/tests/arrays/string_/test_string.py:226]: I think `arr` will work in this file? Else `pd_array`?

File: pandas/tests/arrays/test_datetimelike.py [hunk 4]
```diff
@@ -1160,7 +1160,7 @@ def test_strftime_nat(self):
 
 
 @pytest.mark.parametrize(
-    "array,casting_nats",
+    "arrays,casting_nats",
```
Comment (MarcoGorelli) [inline pandas/tests/arrays/test_datetimelike.py:1163]: how about `ts_array` (for time series array)?

File: pandas/tests/dtypes/test_inference.py [hunk 5]
```diff
@@ -616,9 +616,9 @@ def test_maybe_convert_objects_bool_nan(self):
 
     def test_mixed_dtypes_remain_object_array(self):
         # GH14956
-        array = np.array([datetime(2015, 1, 1, tzinfo=pytz.utc), 1], dtype=object)
```
Comment (MarcoGorelli) [inline pandas/tests/dtypes/test_inference.py:619]: np_array ?

File: pandas/tests/extension/base/getitem.py [hunk 6]
```diff
@@ -324,11 +324,11 @@ def test_take_non_na_fill_value(self, data_missing):
         fill_value = data_missing[1]  # valid
         na = data_missing[0]
 
-        array = data_missing._from_sequence(
+        arrays = data_missing._from_sequence(
```
Comment (MarcoGorelli) [inline pandas/tests/extension/base/getitem.py:327]: I think `arr` will work in this file?

File: pandas/tests/indexing/test_check_indexer.py [hunk 7]
```diff
@@ -26,8 +26,8 @@
     ],
 )
 def test_valid_input(indexer, expected):
-    array = np.array([1, 2, 3])
-    result = check_array_indexer(array, indexer)
+    arrays = np.array([1, 2, 3])
```
Comment (MarcoGorelli) [inline pandas/tests/indexing/test_check_indexer.py:29]: I think `arr` will work in this file too

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
I've made some suggestions for some files, for the others I think `arr` might work (if not, please let me know for which ones it doesn't)

**Comment by deepang17** [comment]
@MarcoGorelli it's ready to be merged. Thank you for your guidance.

File: pandas/tests/series/test_ufunc.py [hunk 8]
```diff
@@ -31,17 +31,17 @@ def arrays_for_binary_ufunc():
 @pytest.mark.parametrize("sparse", SPARSE, ids=SPARSE_IDS)
 def test_unary_ufunc(ufunc, sparse):
     # Test that ufunc(pd.Series) == pd.Series(ufunc)
-    array = np.random.randint(0, 10, 10, dtype="int64")
-    array[::2] = 0
+    u_arr = np.random.randint(0, 10, 10, dtype="int64")
```
Comment (MarcoGorelli) [inline pandas/tests/series/test_ufunc.py:34]: can't this be `arr`? I just tried it and it was fine - besides, what does `u_arr` stand for? Unary array? If so, what is that?

File: pandas/tests/indexing/test_check_indexer.py [hunk 9]
```diff
@@ -26,8 +26,8 @@
     ],
 )
 def test_valid_input(indexer, expected):
-    array = np.array([1, 2, 3])
-    result = check_array_indexer(array, indexer)
+    n_arr = np.array([1, 2, 3])
```
Comment (MarcoGorelli) [inline pandas/tests/indexing/test_check_indexer.py:29]: this looks like it could mean `number of arrays`. `arr` will work here, I just tried it

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Hi @deepang17 

>  it's ready to be merged

I admire your confidence :) Still have some comments though, I think some of these names are unclear and that `arr` should work in those files

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me pending green, cc @jbrockmendel / @jorisvandenbossche for any further comments

File: pandas/tests/arrays/string_/test_string.py [hunk 10]
```diff
@@ -223,31 +223,31 @@ def test_mul(dtype, request):
 
 @pytest.mark.xfail(reason="GH-28527")
 def test_add_strings(dtype):
-    array = pd.array(["a", "b", "c", "d"], dtype=dtype)
+    pd_array = pd.array(["a", "b", "c", "d"], dtype=dtype)
```
Comment (jbrockmendel) [inline pandas/tests/arrays/string_/test_string.py:226]: pd_array -> arr

File: pandas/tests/extension/base/getitem.py [hunk 11]
```diff
@@ -324,11 +324,11 @@ def test_take_non_na_fill_value(self, data_missing):
         fill_value = data_missing[1]  # valid
         na = data_missing[0]
 
-        array = data_missing._from_sequence(
+        d_arr = data_missing._from_sequence(
```
Comment (MarcoGorelli) [inline pandas/tests/extension/base/getitem.py:327]: as suggested in https://github.com/pandas-dev/pandas/pull/40286#discussion_r589747955 , can't this just be `arr`? It's not clear to me what `d_arr` is mean to stand for

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

**Review by MarcoGorelli — APPROVED** [review]

**Review by MarcoGorelli — APPROVED** [review]
I went ahead and renamed the rest of these `*_array` to `arr` and they seem to be fine, not sure why you said there'd be conflicts - perhaps it was just the fixture?

**Comment by deepang17** [comment]
Great, at that time some other variables were also used with the name `arr` and one of them produced an error for me hence I thought it will produce an error for all. But now as you have done it, it seems fine.

**Comment by MarcoGorelli** [comment]
Thanks @deepang17 !

**Pull request merged**